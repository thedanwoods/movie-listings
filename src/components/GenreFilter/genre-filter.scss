.genre-filter {
  padding: 0 0 16px;
}

.genre-filter__heading {
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 700;
  font-size: 13px;
  padding: 0 0 4px 4px;
}
