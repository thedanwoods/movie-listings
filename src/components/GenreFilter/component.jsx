import React from 'react';
import PropTypes from 'prop-types';
import GenreFilterItem from '../GenreFilterItem';

import './genre-filter.scss';

const GenreFilter = ({ genres }) => (
  <div className="genre-filter">
    <h2 className="genre-filter__heading">Genres</h2>
    <ul>
      {genres &&
        genres.map(genre => <GenreFilterItem key={genre.id} genre={genre} />)}
    </ul>
  </div>
);

GenreFilter.propTypes = {
  genres: PropTypes.array.isRequired,
};

export default GenreFilter;
