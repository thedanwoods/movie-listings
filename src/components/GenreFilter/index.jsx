import { connect } from 'react-redux';
import GenreFilter from './component';
import getVisibleMovies from '../../utils/getVisibleMovies';

// Only show genres that are possible based on the current visible movies
export const mapStateToProps = state => {
  const visibleMovies = getVisibleMovies(
    state.movies,
    state.genreFilter,
    state.ratingFilter,
  );
  const visibleGenres = visibleMovies
    .map(movie => movie.genre_ids)
    .reduce((acc, cur) => [...acc, ...cur], []); // Flattens array
  return {
    genres: state.genres.filter(genre =>
      visibleGenres.includes(genre.id),
    ),
  };
};

export default connect(mapStateToProps)(GenreFilter);
