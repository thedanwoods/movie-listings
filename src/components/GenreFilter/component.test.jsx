import React from 'react';
import { shallow } from 'enzyme';
import GenreFilter from './component';

const defaultProps = {
  genres: [{ id: 1 }, { id: 2 }],
};

describe('<GenreFilter /> component', () => {
  it('renders without errors', () => {
    const genreFilter = shallow(<GenreFilter {...defaultProps} />);

    expect(() => genreFilter).not.toThrow();
    expect(genreFilter.debug()).toMatchSnapshot();
  });
});
