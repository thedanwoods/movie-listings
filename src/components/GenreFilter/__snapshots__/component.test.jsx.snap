// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreFilter /> component renders without errors 1`] = `
"<div className=\\"genre-filter\\">
  <h2 className=\\"genre-filter__heading\\">
    Genres
  </h2>
  <ul>
    <Connect(GenreFilterItem) genre={{...}} />
    <Connect(GenreFilterItem) genre={{...}} />
  </ul>
</div>"
`;
